packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -256,6 +254,18 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
   @override
   String get category => FlutterCommandCategory.project;
 
+  // Lookup the default reporter if one was not specified.
+  String _getReporter() {
+    final String? reporter = stringArgDeprecated('reporter');
+    if (reporter != null) {
+      return reporter;
+    }
+    if (globals.platform.environment['GITHUB_ACTIONS'] == 'true') {
```

Comment:
How about case insensitive?

```suggestion
    if (globals.platform.environment['GITHUB_ACTIONS']?.toLowerCase() == 'true') {
```